src/app/shared/operators/set-list-by-batch.ts

```diff
@@ -1,25 +1,29 @@
-import { animationFrames, EMPTY, Observable, of, OperatorFunction, pipe } from 'rxjs';
-import { map, pairwise, scan, startWith, switchMap, takeUntil, takeWhile } from 'rxjs/operators';
+import { animationFrames, EMPTY, Observable, ObservableInput, OperatorFunction, pipe } from 'rxjs';
+import { map, scan, switchMap, take, takeUntil } from 'rxjs/operators';
 
 interface Options {
   size?: number,
-  until?: Observable<any>,
+  until?: ObservableInput<any>,
+  interval?: () => Observable<unknown>,
 }
-export function setListByBatch<T>({ size = 25, until = EMPTY }: Options): OperatorFunction<T[], T[]> {
+export function setListByBatch<T>({ size = 25, until = EMPTY, interval = animationFrames }: Options): OperatorFunction<T[], T[]> {
```

Comment:
You probably need a creation operation (which transform a list to an observable) and not a pipe operator so that you can use it directly in a switch map in the component. You could do a pipe operator but then you would need to do a `of(list).pipe(...)` to use it, which would be less clean IMO (which is not what you did as you use two different variables anyway)

Reply:
It is quite unexpected for a pipe operator to set something. (setListByBatch) (-> the naming is probably not appropriate)

Reply:
Fixed